Extensions/Mongo/Cosmos.DataTransfer.MongoExtension/MongoDataSourceExtension.cs

```diff
@@ -27,30 +28,95 @@ public async IAsyncEnumerable<IDataItem> ReadAsync(IConfiguration config, ILogge
 
             foreach (var collection in collectionNames)
             {
-                await foreach (var item in EnumerateCollectionAsync(context, collection, logger).WithCancellation(cancellationToken))
+                await foreach (var item in EnumerateCollectionAsync(context, collection, settings.Query, logger).WithCancellation(cancellationToken))
                 {
                     yield return item;
                 }
             }
         }
     }
 
-    public async IAsyncEnumerable<IDataItem> EnumerateCollectionAsync(Context context, string collectionName, ILogger logger)
+    public async IAsyncEnumerable<IDataItem> EnumerateCollectionAsync(Context context, string collectionName, string? query, ILogger logger)
     {
         logger.LogInformation("Reading collection '{Collection}'", collectionName);
         var collection = context.GetRepository<BsonDocument>(collectionName);
         int itemCount = 0;
-        foreach (var record in await Task.Run(() => collection.AsQueryable()))
+
+        IAsyncEnumerable<BsonDocument> documents;
+        
+        if (!string.IsNullOrWhiteSpace(query))
+        {
+            logger.LogInformation("Applying query filter to collection '{Collection}': {Query}", collectionName, query);
+            documents = GetQueryDocumentsAsync(collection, query, collectionName, logger);
+        }
+        else
+        {
+            // Use existing queryable approach when no filter is specified
```

Comment:
Let's add a logging line here indicating that there is no query supplied. That makes it helpful when debugging and knowing what's going on.